openff/toolkit/tests/test_topology.py

```diff
@@ -1448,3 +1451,116 @@ def test_impossible_error(self):
     def test_no_key(self):
         with pytest.raises(ValueError, match="not in possible"):
             ImproperDict.index_of([0, 1, 2, 4], possible=[(4, 1, 2, 0)])
+
+
+@pytest.mark.parametrize(
+    "generate_positions",
+    [
+        # Test an array
+        lambda n, rng: rng.random((n, 3)),
+        # Test a list of lists
+        lambda n, rng: [[rng.random(), rng.random(), rng.random()] for _ in range(n)],
+        # Test a list of tuples
+        lambda n, rng: [(rng.random(), rng.random(), rng.random()) for _ in range(n)],
+    ],
+)
```

Comment:
Now uses range/arange instead of rng. I also moved the generate_positions parameters to the test in which its actually necessary, rather than the whole class.